models/servicenow__ticket_enhanced.sql

```diff
@@ -235,8 +238,8 @@ select
   change_request.is_change_request_unauthorized,
   task.source_relation
 
-from task
-left join problem_task
+from task 
+left join problem_task 
```

Comment:
You put a lot of effort into understanding how tasks, problems, incidents, and change requests relate to one another. I worry we will ultimately forget how these relationships work and I also feel we made a pretty opinionated stance on choosing task to be the grain of the primary model (correct me if I am wrong there).

I feel it would be beneficial if you could include a DECISIONLOG.md and include your own documentation around how the relationships of these ServiceNow components work. This way customers will be able to reference this and understand the decision behind the grain of the model, and this will be beneficial for future package maintenance for us to look back on to understand how all the ServiceNow components relate to one another.

Reply:
good idea Joe! Added this